cuda.graph_graph_color* COLORING_VBD test failures with cuda/9.2 + gcc/7.2 on White

The Jenkins job `KokkosKernels_White_CudaOpenMP_cuda_92_gcc_720` shows at least the following test failure:

`00:03:47 [  FAILED  ] cuda.graph_graph_color_double_int_int_TestExecSpace`

Subsequent tests were also failing in the Jenkins job, but this may be due to the first test failure. After reproducing and filtering out the first failing test I'll post other valid failures.

This began failing after updating the `test_all_sandia` script to match the versions of cuda and gcc tested by Kokkos and the Trilinos ATDM scripts. 

Edit: Added the name of the Jenkins job.

The Jenkins job is [KokkosKernels_White_CudaOpenMP_cuda_92_gc_720][1] ... with [build-time trend][2].

It looks like from 8/24 until 10/2 the execution time had been around 10 seconds -- which indicates the tests weren't actually running (even though the Jenkins job returned `SUCCESS`).

Starting on 10/3 the execution times have been in the 24 minute range, but from 7/12 to 8/24 the runs were typically taking 7h 0m and timing out in the queue so they never really ran.

My read on this is that this test hadn't really been run between 7/15 and 10/2. :( 

Looking at the [console output][3] from last night's run, there is a lot more than just the 1 failing test shown above:
```
00:03:47 
00:03:47 113 FAILED TESTS
00:03:47 terminate called after throwing an instance of 'std::runtime_error'
00:03:47   what():  cudaMemcpy( dst , src , n , cudaMemcpyDefault ) error( cudaErrorIllegalAddress): an illegal memory access was encountered /home/jenkins/white/workspace/KokkosKernels_White_CudaOpenMP_cuda_92_gcc_720/kokkos/core/src/Cuda/Kokkos_CudaSpace.cpp:89
00:03:47 Traceback functionality not available
00:03:47 
00:03:47 make[2]: *** [test-cuda] Aborted (core dumped)
00:03:47 make[2]: Leaving directory `/home/jenkins/white/workspace/KokkosKernels_White_CudaOpenMP_cuda_92_gcc_720/TestAll_2018-10-07_23.40.01/cuda/9.2.88/OpenMP_Cuda-release/unit_test'
00:03:47 make[1]: *** [test] Error 2
00:03:47 make[1]: Leaving directory `/home/jenkins/white/workspace/KokkosKernels_White_CudaOpenMP_cuda_92_gcc_720/TestAll_2018-10-07_23.40.01/cuda/9.2.88/OpenMP_Cuda-release/unit_test'
00:03:47 make: *** [test] Error 2
00:03:48 #######################################################
00:03:48 PASSED TESTS
00:03:48 #######################################################
00:03:48 #######################################################
00:03:48 FAILED TESTS
00:03:48 #######################################################
00:03:48 cuda-9.2.88-OpenMP_Cuda-release (test failed)
```

but they are all on that cudaMalloc instruction as far as I can tell:

```
00:03:47 [ RUN      ] cuda.batched_scalar_team_gemm_t_t_dcomplex_dcomplex
00:03:47 unknown file: Failure
00:03:47 C++ exception with description "cudaMalloc( &ptr, arg_alloc_size ) error( cudaErrorIllegalAddress): an illegal memory access was encountered /home/jenkins/white/workspace/KokkosKernels_White_CudaOpenMP_cuda_92_gcc_720/kokkos/core/src/Cuda/Kokkos_CudaSpace.cpp:175
00:03:47 Traceback functionality not available
00:03:47 " thrown in the test body.
00:03:47 [  FAILED  ] cuda.batched_scalar_team_gemm_t_t_dcomplex_dcomplex (0 ms)
```

@ndellingwood Tribal knowledge question: How are the tests driven by KokkosKernels, does it generate 1 mongo test-everything binary with all the tests dumped in so a failure on one can affect the others, or does it generate separate test binaries by package or by test so a failure in one won't disrupt others?  Normally I'd think that if I see 113 test fails then it'd imply something in the core library is suspect -- but it sounds like in KokkosKernels the testing is designed to make the single mongo-binary that includes everything so one test failing can affect others and make them fail even if they're not dependent on the thing that's failing?

[1]: https://jenkins-son.sandia.gov/view/KokkosKernels/job/KokkosKernels_White_CudaOpenMP_cuda_92_gcc_720/
[2]: https://jenkins-son.sandia.gov/view/KokkosKernels/job/KokkosKernels_White_CudaOpenMP_cuda_92_gcc_720/buildTimeTrend
[3]: https://jenkins-son.sandia.gov/view/KokkosKernels/job/KokkosKernels_White_CudaOpenMP_cuda_92_gcc_720/401/console

--- a/src/graph/KokkosGraph_graph_color.hpp
+++ b/src/graph/KokkosGraph_graph_color.hpp
@@ -100,7 +100,7 @@ void graph_color_symbolic(
   case COLORING_VBD:
   case COLORING_VBDBIT:
     typedef typename Impl::GraphColor_VBD <typename KernelHandle::GraphColoringHandleType, lno_row_view_t_, lno_nnz_view_t_> VBDGraphColoring;
-    gc = new VBDGraphColoring(num_rows, entries.dimension_0(), row_map, entries, gch);
+    gc = new VBDGraphColoring(num_rows, entries.extent(0), row_map, entries, gch);
     break;
 
   case COLORING_EB:

--- a/unit_test/graph/Test_Graph_graph_color.hpp
+++ b/unit_test/graph/Test_Graph_graph_color.hpp
@@ -130,16 +130,17 @@ void test_coloring(lno_t numRows,size_type nnz, lno_t bandwidth, lno_t row_size_
                                                        , COLORING_VBBIT 
                                                        , COLORING_VBCS 
                                                        , COLORING_EB
-                                                       , COLORING_VBDBIT
                                                        };
 
   #ifdef KOKKOS_ENABLE_CUDA
   if( !std::is_same< typename device::execution_space, Kokkos::Cuda >::value )
   {
     coloring_algorithms.push_back(COLORING_VBD);
+    coloring_algorithms.push_back(COLORING_VBDBIT);
   }
   #else
   coloring_algorithms.push_back(COLORING_VBD);
+  coloring_algorithms.push_back(COLORING_VBDBIT);
   #endif
 
   for (size_t ii = 0; ii < coloring_algorithms.size(); ++ii) {

--- a/unit_test/graph/Test_Graph_graph_color.hpp
+++ b/unit_test/graph/Test_Graph_graph_color.hpp
@@ -124,9 +124,24 @@ void test_coloring(lno_t numRows,size_type nnz, lno_t bandwidth, lno_t row_size_
   graph_t static_graph (sym_adj, sym_xadj);
   input_mat = crsMat_t("CrsMatrix", numCols, newValues, static_graph);
 
-  ColoringAlgorithm coloring_algorithms[] = {COLORING_DEFAULT, COLORING_SERIAL, COLORING_VB, COLORING_VBBIT, COLORING_VBCS, COLORING_EB, COLORING_VBD, COLORING_VBDBIT};
+  std::vector<ColoringAlgorithm> coloring_algorithms = { COLORING_DEFAULT, 
+                                                         COLORING_SERIAL, 
+                                                         COLORING_VB, 
+                                                         COLORING_VBBIT, 
+                                                         COLORING_VBCS, 
+                                                         COLORING_EB, 
+                                                         COLORING_VBDBIT };
 
-  for (int ii = 0; ii < 8; ++ii){
+#ifdef KOKKOS_ENABLE_CUDA
+  if( !std::is_same< typename device::execution_space, Kokkos::Cuda >::value )
+  {
+    coloring_algorithms.push_back(COLORING_VBD);
+  }
+#else
+  coloring_algorithms.push_back(COLORING_VBD);
+#endif
+
+  for (size_t ii = 0; ii < coloring_algorithms.size(); ++ii){
     ColoringAlgorithm coloring_algorithm = coloring_algorithms[ii];
     color_view_t vector_colors;
     size_t num_colors;

--- a/unit_test/graph/Test_Graph_graph_color.hpp
+++ b/unit_test/graph/Test_Graph_graph_color.hpp
@@ -124,24 +124,25 @@ void test_coloring(lno_t numRows,size_type nnz, lno_t bandwidth, lno_t row_size_
   graph_t static_graph (sym_adj, sym_xadj);
   input_mat = crsMat_t("CrsMatrix", numCols, newValues, static_graph);
 
-  std::vector<ColoringAlgorithm> coloring_algorithms = { COLORING_DEFAULT, 
-                                                         COLORING_SERIAL, 
-                                                         COLORING_VB, 
-                                                         COLORING_VBBIT, 
-                                                         COLORING_VBCS, 
-                                                         COLORING_EB, 
-                                                         COLORING_VBDBIT };
+  std::vector<ColoringAlgorithm> coloring_algorithms = { COLORING_DEFAULT 
+                                                       , COLORING_SERIAL 
+                                                       , COLORING_VB 
+                                                       , COLORING_VBBIT 
+                                                       , COLORING_VBCS 
+                                                       , COLORING_EB
+                                                       , COLORING_VBDBIT
+                                                       };
 
-#ifdef KOKKOS_ENABLE_CUDA
+  #ifdef KOKKOS_ENABLE_CUDA
   if( !std::is_same< typename device::execution_space, Kokkos::Cuda >::value )
   {
     coloring_algorithms.push_back(COLORING_VBD);
   }
-#else
+  #else
   coloring_algorithms.push_back(COLORING_VBD);
-#endif
+  #endif
 
-  for (size_t ii = 0; ii < coloring_algorithms.size(); ++ii){
+  for (size_t ii = 0; ii < coloring_algorithms.size(); ++ii) {
     ColoringAlgorithm coloring_algorithm = coloring_algorithms[ii];
     color_view_t vector_colors;
     size_t num_colors;

--- a/unit_test/graph/Test_Graph_graph_color_deterministic.hpp
+++ b/unit_test/graph/Test_Graph_graph_color_deterministic.hpp
@@ -151,15 +151,17 @@ void test_coloring_deterministic(lno_t numRows, size_type nnz) {
   graph_t static_graph (adj, xadj);
   crsMat_t input_mat("CrsMatrix", numCols, newValues, static_graph);
 
-  std::vector<ColoringAlgorithm> coloring_algorithms = { COLORING_VBDBIT };
+  std::vector<ColoringAlgorithm> coloring_algorithms;
 
   #ifdef KOKKOS_ENABLE_CUDA
   if( !std::is_same< typename device::execution_space, Kokkos::Cuda >::value )
   {
-    coloring_algorithms.push_back(COLORING_VBD);
+     coloring_algorithms.push_back(COLORING_VBD);
+     coloring_algorithms.push_back(COLORING_VBDBIT);
   }
   #else
   coloring_algorithms.push_back(COLORING_VBD);
+  coloring_algorithms.push_back(COLORING_VBDBIT);
   #endif
 
   for (size_t ii = 0; ii < coloring_algorithms.size(); ++ii) {
@@ -184,7 +186,7 @@ void test_coloring_deterministic(lno_t numRows, size_type nnz) {
     }
 
     EXPECT_TRUE( (num_conflict == 0));
-  //device::execution_space::finalize();
+    //device::execution_space::finalize();
 
   }
 

--- a/unit_test/graph/Test_Graph_graph_color_deterministic.hpp
+++ b/unit_test/graph/Test_Graph_graph_color_deterministic.hpp
@@ -151,9 +151,18 @@ void test_coloring_deterministic(lno_t numRows, size_type nnz) {
   graph_t static_graph (adj, xadj);
   crsMat_t input_mat("CrsMatrix", numCols, newValues, static_graph);
 
-  ColoringAlgorithm coloring_algorithms[] = {COLORING_VBD, COLORING_VBDBIT};
+  std::vector<ColoringAlgorithm> coloring_algorithms = { COLORING_VBDBIT };
 
-  for (int ii = 0; ii < 2; ++ii){
+  #ifdef KOKKOS_ENABLE_CUDA
+  if( !std::is_same< typename device::execution_space, Kokkos::Cuda >::value )
+  {
+    coloring_algorithms.push_back(COLORING_VBD);
+  }
+  #else
+  coloring_algorithms.push_back(COLORING_VBD);
+  #endif
+
+  for (size_t ii = 0; ii < coloring_algorithms.size(); ++ii) {
     ColoringAlgorithm coloring_algorithm = coloring_algorithms[ii];
     color_view_t vector_colors;
     size_t num_colors;

